bundles/framework/timeseries/service/TimeseriesLayerService.js

```diff
@@ -0,0 +1,146 @@
+                }
+            }
+        },
+        /**
+         * @method _checkMultipleLayers
+         * @private
+         * Show popup if there are more than one selected timeseries layers
+         */
+        _checkMultipleLayers: function () {
+            if (this._timeseriesService.getCountByType('layer') > 1) {
+                this._popupService.closeAllPopups(false);
+                var popup = this._popupService.createPopup();
+                var closeBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+                closeBtn.setTitle(this.loc('alert.ok'));
+                closeBtn.setHandler(function () {
+                    popup.close(true);
+                });
+                popup.show(this.loc('alert.title'), this.loc('alert.message'), [closeBtn]);
+            }
+        },
+        /**
+         * @public @method onEvent
+         * Event is handled forwarded to correct eventHandlers if found or discarded if not.
+         * @param {Oskari.mapframework.event.Event} event a Oskari event object
+         */
+        onEvent: function (event) {
+            var handler = this.__eventHandlers[event.getName()];
+            if (!handler) {
+                return;
+            }
+            return handler.apply(this, [event]);
+        },
+        /**
+         * @method registerLayerType
+         * Registers animator/delegate implementation for certain type of layer
+         * @param {String} type layer type (AbstractLayer.getLayerType())
+         * @param {Function} factory function that takes layerId as parameter and returns instance that implements Oskari.mapframework.bundle.timeseries.TimeseriesDelegateProtocol
+         */
+        registerLayerType: function (type, factory) {
+            this._layerTypeAnimators[type] = factory;
+        },
+        /**
+         * @method _layerDelegateFactory
+         * @private
+         * Requests change in current selected time
+         * @param {String} layerId id of layer
+         * @param {String} layerType type of layer (AbstractLayer.getLayerType())
+         * @return {Oskari.mapframework.bundle.timeseries.TimeseriesDelegateProtocol} doneCallback callback that will be called after new time has been loaded
+         */
+        _layerDelegateFactory: function (layerId, layerType) {
+            var factory = this._layerTypeAnimators[layerType];
+            if (!factory) {
+                this._log.warn('No animator defined for layer type "' + layerType + '"!');
+                return;
+            }
+            return factory(layerId);
+        },
+        /**
+         * @method updateTimeseriesLayers
+         * Update timeseries service state based on current selected layers
+         */
+        updateTimeseriesLayers: function () {
+            var me = this,
+                layers = me._sandbox.findAllSelectedMapLayers();
+            for (var i = 0; i < layers.length; i++) {
+                if (layers[i].hasTimeseries()) {
```

Comment:
I prefer using continue when all the code inside a loop is in a single if statement to prevent having large blocks and/or }}} kinds of rows, but it's mostly an matter of opinion.

    if (!layers[i].hasTimeseries()) {
         continue;
    }